src/io/sampling.jl

```diff
@@ -585,36 +562,62 @@ function set_factor_bounds!(d::Domain; factors...)::Nothing
     return nothing
 end
 
-_unzip(a) = map(x -> getfield.(a, x), fieldnames(eltype(a)))
+function _continuous_bounds(dom::Domain, factor::Symbol, new_bounds::Tuple)::Tuple
+    new_lower, new_upper = new_bounds[1:2]
+    new_val::Float64 = new_lower + 0.5 * (new_upper - new_lower)
+    return (new_bounds, new_val)
+end
+
+function _discrete_bounds(dom::Domain, factor::Symbol, new_bounds::Tuple)::Tuple
+    new_lower, new_upper = new_bounds[1:2]
+    (new_lower % 1.0 == 0.0 && new_upper % 1.0 == 0.0) ||
+        @warn "Upper and/or lower bounds for discrete variables should be integer numbers."
+
+    new_lower = round(new_lower)
+    new_upper = min(round(new_upper) + 1.0, get_default_bounds(dom, factor)[2])
+    new_bounds = (new_lower, new_upper)
+
+    new_val::Int64 = floor(new_lower + 0.5 * (new_upper - new_lower))
+
+    return (new_bounds, new_val)
+end
 
 """
-    function _check_bounds_range(new_bounds::Tuple, factor::Symbol, dom::Domain)
+    _check_bounds_range(dom::Domain, factor::Symbol, new_bounds::Tuple)::Nothing
 
-    Check new parameter bounds are within default parameter bounds
+Check new parameter bounds are within default parameter bounds
 """
-function _check_bounds_range(
-    new_bounds::Tuple, default_bounds::Tuple, factor::Symbol, dom::Domain
-)::Nothing
-    out_of_bounds::Bool = false
+function _check_bounds_range(dom::Domain, factor::Symbol, new_bounds::Tuple)::Nothing
+    default_lower, default_upper = get_default_bounds(dom, factor)
     new_lower, new_upper = new_bounds
-    default_lower, default_upper = default_bounds
+
+    out_of_bounds::Bool = false
 
     if _is_discrete_factor(dom, factor)
         # If factor is discrete, sampled values must be within default_lower and (default_upper - 1)
-        out_of_bounds = (new_lower < default_lower) || (new_upper > (default_upper - 1))
+        out_of_bounds = (new_lower < default_lower) || (new_upper > default_upper)
     else
         out_of_bounds = (new_lower < default_lower) || (new_upper > default_upper)
     end
 
+    # Check if new bounds are withing default range
```

Comment:
"withing" -> "within"

add "the"

`# Check if new bounds are within the default range`